tests/baselines/reference/checkJsxChildrenProperty14.js

```diff
@@ -0,0 +1,85 @@
+//// [file.tsx]
+import React = require('react');
+
+interface Prop {
+    a: number,
+    b: string,
+    children: JSX.Element | JSX.Element[];
+}
+
+class Button extends React.Component<any, any> {
+    render() {
+        return (<div>My Button</div>)
+    }
+}
+
+function AnotherButton(p: any) {
+    return <h1>Just Another Button</h1>;
+}
+
+function Comp(p: Prop) {
+    return <div>{p.b}</div>;
+}
+
+// OK
+let k1 = <Comp a={10} b="hi"><></><Button /><AnotherButton /></Comp>;
+let k2 = <Comp a={10} b="hi"><><Button /></><AnotherButton /></Comp>;
+let k3 = <Comp a={10} b="hi"><><Button /><AnotherButton /></></Comp>;
+
+interface SingleChildProp {
+    a: number,
+    b: string,
+    children: JSX.Element;
+}
+
+function SingleChildComp(p: SingleChildProp) {
+    return <div>{p.b}</div>;
+}
+
+// OK
+let k4 = <SingleChildComp a={10} b="hi"><><Button /><AnotherButton /></></SingleChildComp>;
```

Comment:
I'd kinda prefer for these to be a different type like, say, `JSX.Fragment`, but having one would not be useful anyway without nominal typing.

Reply:
We'd like to make `JSX.Element` generic in the future - possibly looking it up and instantiating it based on the return type of the factory function; that would handle the `Fragment` case somewhat elegantly.

Reply:
Generic JSX.Element<T> would be great. Outside of the React bubble, TypeScript-oriented developers like myself often wonder why the disconnect.